hydra-node/test/Hydra/Chain/Direct/Contract/Close.hs

```diff
@@ -207,50 +251,41 @@ genCloseMutation (tx, _utxo) =
             , ChangeInput
                 healthyOpenHeadTxIn
                 (replacePolicyIdWith Fixture.testPolicyId otherHeadId healthyOpenHeadTxOut)
-                (Just $ toScriptData healthyCloseDatum)
+                (Just $ toScriptData healthyOpenHeadDatum)
             ]
     ]
  where
   headTxOut = fromJust $ txOuts' tx !!? 0
 
-  closeRedeemer snapshotNumber sig =
-    Head.Close
-      { snapshotNumber = toInteger snapshotNumber
-      , signature = toPlutusSignatures sig
-      , utxoHash = ""
-      }
-
   mutateCloseUTxOHash :: Gen (TxOut CtxTx)
   mutateCloseUTxOHash = do
     mutatedUTxOHash <- genHash
-    pure $ changeHeadOutputDatum (mutateHash mutatedUTxOHash) headTxOut
-
-  mutateHash mutatedUTxOHash = \case
-    Head.Closed{snapshotNumber, parties, contestationDeadline, headId} ->
-      Head.Closed
-        { snapshotNumber
-        , utxoHash = toBuiltin mutatedUTxOHash
-        , parties
-        , contestationDeadline
-        , headId
-        }
-    st -> error $ "unexpected state " <> show st
-  -- In case contestation period param is 'Nothing' we will generate arbitrary value
-  mutateClosedContestationDeadline :: Integer -> Gen (TxOut CtxTx)
-  mutateClosedContestationDeadline contestationPeriodSeconds = do
-    -- NOTE: we need to be sure the generated contestation period is large enough to have an impact on the on-chain
-    -- deadline computation, which means having a resolution of seconds instead of the default picoseconds
-    pure $ changeHeadOutputDatum (mutateContestationDeadline contestationPeriodSeconds) headTxOut
-
-  mutateContestationDeadline contestationPeriod = \case
-    Head.Closed{snapshotNumber, utxoHash, parties} ->
-      Head.Closed
-        { snapshotNumber
-        , utxoHash
-        , parties
-        , contestationDeadline =
-            let closingTime = slotNoToUTCTime healthySlotNo
-             in posixFromUTCTime $ addUTCTime (fromInteger contestationPeriod) closingTime
-        , headId = toPlutusCurrencySymbol Fixture.testPolicyId
-        }
-    st -> error $ "unexpected state " <> show st
+    pure $ changeHeadOutputDatum (replaceUtxoHash $ toBuiltin mutatedUTxOHash) headTxOut
+
+data CloseInitialMutation
+  = MutateCloseContestationDeadline'
```

Comment:
I actually I think we should test all the other mutations here since we do not  randomly generate either a _normal_ healthy close or an _initial_ healthy close. I'll try to explain that in the code and let as is for this P.R.